express-notes-api/index.js

```diff
@@ -0,0 +1,114 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const notesJson = require('./data.json');
+
+const writingFile = () => {};
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const gradesArray = [];
+  if (notesJson.notes) {
+    for (const notes in notesJson.notes) {
+      gradesArray.push(notesJson.notes[notes]);
+    }
+  } else if (!notesJson.notes) {
+    res.json([]);
+  }
+  res.json(gradesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  let noteGet = {};
+  const id = +req.params.id;
+
+  if (notesJson.notes[id]) {
+    noteGet = notesJson.notes[id];
+    res.status(200);
+  } else if (id < 0 || !Number.isInteger(id) || Number.isNaN(id)) {
+    noteGet.error = 'Id must be a positive integer';
+    res.status(400);
+  } else if (!notesJson.notes[id]) {
+    noteGet.error = `Error: cannot find note with id ${id}`;
+    res.status(404);
+  }
+
+  res.json(noteGet);
+});
+
+app.post('/api/notes', (req, res) => {
+  const content = req.body.content;
+  let newNote = {};
+  if (content === undefined || content === '') {
+    newNote.error = 'content is a require field';
+    res.status(400).json(newNote);
+  } else if (content) {
+    const id = notesJson.nextId++;
+    newNote = {
+      id,
+      content
+    };
+    notesJson.notes[id] = newNote;
+    fs.writeFile('./data.json', JSON.stringify(notesJson), err => {
+      if (err) {
+        console.error('An unexpected error occurred');
+        res.status(500).json({ error: 'An unexpected error occurred.' });
+      } else {
+        res.status(201).json(newNote);
+      }
+    });
+  }
+});
+
+app.delete('/api/notes/:id', (req, res) => {
+  const deletedNote = {};
+  const id = Math.trunc(+req.params.id);
```

Comment:
This will fix the `id` if it is a decimal, you don't want to fix client errors for them. You just want to let them know if they send an invalid `id`. Here just doing `const id = +req.params.id` or `const id = Number(req.params.id)`, either would be fine. The important part is that we aren't fixing the clients errors for them, they could be sending a decimal on accident so we want to let them know.